codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/protocol/ServerProtocolTestGenerator.kt

```diff
@@ -343,6 +338,34 @@ class ServerProtocolTestGenerator(
         }
     }
 
+    private fun checkBody(rustWriter: RustWriter, body: String, mediaType: String?) {
+        rustWriter.rustTemplate(
+            """
+            let body = #{Hyper}::body::to_bytes(http_response.into_body()).await.expect("unable to extract body to bytes");
+            """,
+            *codegenScope
+        )
+        if (body == "") {
+            rustWriter.rustTemplate(
+                """
+                // No body
+                #{AssertEq}(std::str::from_utf8(&body).unwrap(), "");
```

Comment:
```suggestion
                #{AssertEq}(std::str::from_utf8(&body).expect("Should be an empty body string"), "");
```

Reply:
Not sure if the comment is the right one, but I would use `expect()` instead of `unwrap()`

Reply:
This will panic if the body is not UTF8, not if it's empty.

`unwrap()` is fine here because it's a test, and `from_utf8()` already returns an `Err` with a reason as to why the body is not UTF8.

Reply:
Ok, makes sense.